File: my-folder/0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max_so_far = Integer.MIN_VALUE;
        int max_ending_here = 0;
        for (int num : nums) {
            max_ending_here += num;
            max_so_far = Math.max(max_so_far, max_ending_here);
            if (max_ending_here < 0) {
                max_ending_here = 0;
            }
        }
        return max_so_far;
    }
}

File: my-folder/0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        # We will keep track of the current sum
        # If the current sum < 0, set it to 0
        res = nums[0]
        curr = 0
        for i in range(len(nums)):
            curr += nums[i]
            res = max(res, curr)
            if curr < 0:
                curr = 0
        return res
